### ISSUE TITLE
Performance problems when change list is large

### ISSUE THREAD

**Issue by lukaville** [badoo/codeisok#86]
When using treediff and if the number of changed files is about ~1k there are noticeable problems with performance: the frame rate is very low, click actions are executed with a significant delay.

Not sure what the cause of the issues: file tree or the contents (or both). Probably, we should implement some mechanisms for lazy loading and add only visible elements to DOM.

**Comment by NikhilVerma** [comment]
@lukaville  I agree, it can get extremely slow for large diffs. I'll look into improvements from the UI side.
